src/index.ts

```diff
@@ -19,10 +19,29 @@ const displayStackSimple = winston.format((info: TransformableInfo) => {
   return info
 })
 
+const addIndent = (str: string, indentLevel: number): string => {
+  return str.split("\n").map(item => `${'  '.repeat(indentLevel)}${item}`).join("\n")
+}
+
+const mergeStacks = (err: any) => {
+  if (!(err.message instanceof Error)) {
+    return ''
+  }
+  let error = err.message
+  let stack = `${error.stack}`
+  let indent = 1
+  while (error && error.error && error.error instanceof Error && error.error.stack) {
+    stack = `${stack}\n`
+    error = error.error
+    stack = `${stack}${addIndent(error.stack, indent++)}\n`
+  }
+  return stack
+}
+
 const displayStackJson = winston.format((info: any) => {
   if (info.level === 'error' && info.message && info.message instanceof Error) {
     return Object.assign({}, info, {
-      stack: info.message.stack,
+      stack: mergeStacks(info),
```

Comment:
At this point it is known that `info.message` is of type `Error` and it's the only property of `info` object that is used inside of `mergeStacks`. Knowing this could the `mergeStacks` signature be adjusted to `mergeStacks(err: Error): string` adding more type safety?

Reply:
Good point in this case we are using on implementation side nested error objects with ts-custom-errors, they just extend Error. When activating more strictness here, `error.error` is invalid Type as Errors do not have nested errors.